### ISSUE TITLE
Webpack ignores changes to vendor.scss

### ISSUE THREAD

**Issue by xetys** [jhipster/generator-jhipster#5039]
I generated a [sample setup](https://github.com/xetys/jhipster4-uaa-setup) with sass enabled.

However changing thinks inside the sass files like global.scss, home.scss etc. works like a charm, webpack auto reloads the browser and changes are visible. However this doesn't work to vendor.scss for some reason.

I didn't find the cause, yet. Just writing down what I find

*edit*: the changes finally take effect, if you stop yarn, delete ./build/www and start yarn again

**Comment by jdubois** [comment]
I don't think it should reload those files: the "vendor" files come from third party libs, you should extend them, but not modify them.

**Comment by deepu105** [comment]
Did you add a new library in vendor.scss?

**Comment by deepu105** [comment]
you could also try to execute `yarn run webpack:build` so that vendor files are recompiled. If you installed the lib via npm/yarn then the postInstall should have taken care of it

**Comment by xetys** [comment]
I installed angular/material and started to add materials specific sass files into the vendor.

The question is, if its ok to force users to rebuild with yarn on vendor code changes, even it generally shouldn't happen that often

Maybe I am misunderstanding the usage of the vendor.css / vendor.scss, but for me it's like

* global.(s)css: I need some app wide css class, related to my project
* vendor.(s)css: I need to adjust some vendor libraries, as bootstrap, fontawesome, theming, angular etc..

to have a clear distinction between app styles and vendor extends.

**Comment by jdubois** [comment]
For me the "vendor" are installed through Yarn/NPM and should not be
touched.
What you customize is inside "app".

Le 31 janv. 2017 11:31 PM, "David Steiman" <notifications@github.com> a
écrit :

**Comment by deepu105** [comment]
Yes your custom css even if its to override something in a vendor lib
should go to global.scss/css


Thanks & Regards,
Deepu

**Comment by gmarziou** [comment]
Maybe we should add 2 lines of comments to these files to guide people on their purpose.

**Comment by jdubois** [comment]
@gmarziou we can't, they are provided by Yarn.
For me it's quite logical that "vendor" is provided by third-party tools, so you shouldn't touch them.

I'm closing this because I don't think there is any issue here, just a misunderstanding.

**Comment by xetys** [comment]
AFAIK they are not provided by yarn....the existing lines are provided from our generator, and not yarn

see https://github.com/jhipster/generator-jhipster/blob/master/generators/client/templates/angular/src/main/webapp/content/scss/_vendor.scss

or https://github.com/jhipster/generator-jhipster/blob/master/generators/client/templates/angular/src/main/webapp/content/css/_vendor.css

(by posting this I see the SASS template doesn't use webpacks convention of using ~ instead of "node_modules"...we should fix that)

both global and vendor.(s)css are treated similary in webpack common...

in other words, if our generator doesn't generate these files as they are, we would have bootstrap and font-awesome disabled. They won't get added from yarn. Or at least: I don't know how this should work, I already added some different stuff and nothing of this took effect on the vendor.scss.

I'm not sure this is just a misunderstanding, as I won't be the first one who tries to modify that file and notice that issue.

**Comment by kaidohallik** [comment]
Webpack development server is not tracking vendor files.  
Starting gradle or maven with `webpack` profile doesn't change vendor files (both are using `webpack:build:dev`, not `webpack:build`).  
So I made PR where I added instructions, how to get vendor changes in.  
Alternative solution is to add this instruction to the `README.md` at the end of the vendor section.
